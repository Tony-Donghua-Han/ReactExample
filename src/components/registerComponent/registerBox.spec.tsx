import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import RegisterBox from "./registerBox";

describe("LoginForm", (): void => {
    let wrapper: ShallowWrapper;
    beforeEach((): void => {
        wrapper = shallow(<RegisterBox />);
    });

    test("should render without crushing", (): void => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    test("should render title text Create an account", (): void => {
        expect(
            wrapper
                .find(".title")
                .first()
                .text(),
        ).toBe("Create an account");
    });

    test("should render footer with Already On Financhy? and Login In link", (): void => {
        const footer: ShallowWrapper = wrapper.find(".footer");
        expect(footer.childAt(0).text()).toBe("Already on Financhy?");
    });

    test("should render footer with Login In link", (): void => {
        const footer: ShallowWrapper = wrapper.find(".footer");
        expect(footer.find(".link").text()).toBe("Login Ins");
    });
});
