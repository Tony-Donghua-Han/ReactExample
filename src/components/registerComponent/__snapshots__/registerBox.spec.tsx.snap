// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render without crushing 1`] = `
"<div className=\\"registerBox\\" id=\\"registerBox\\">
  <div className=\\"title\\">
    Create an account
  </div>
  <div className=\\"divider\\" />
  <div className=\\"form\\">
    <RegistrationForm onSubmit={[Function: onSubmit]} />
  </div>
  <div className=\\"titleWithLine\\">
    or
  </div>
  <div className=\\"footer\\">
    <div>
      Already on Financhy?
    </div>
    <div className=\\"link\\">
      Login In
    </div>
  </div>
</div>"
`;
